python_modules/dagster/dagster/_core/definitions/data_version.py

```diff
@@ -43,7 +43,10 @@ class UnknownValue:
 class DataVersion(
     NamedTuple(
         "_DataVersion",
-        [("value", str)],
+        [
+            ("value", str),
+            ("data_time", Optional[float]),
```

Comment:
I think we'll want to put the data time on whatever encloses the `DataVersion`, e.g. `ObserveResult`.  Users who aren't dealing with the versioning system shouldn't need to deal with the data version concept.

Reply:
Agreed. Should you be able to emit `AssetObservation`s from `@observable_source_asset`s? API is a bit odd with the system tags but we could change the  \_\_new\_\_ params

Reply:
I don't think so, for the same reason we don't expect users to emit `AssetMaterialization`s from `@asset`s. We use `MaterializeResult` instead, so the analog would be `ObserveResult`.